R/restrict_derivation.R

```diff
@@ -78,11 +78,8 @@ restrict_derivation <- function(dataset,
                                 filter) {
   # Check input
   assert_data_frame(dataset)
-  assert_function(derivation, params = c("dataset"))
+  assert_function(derivation)
   assert_s3_class(args, "params", optional = TRUE)
-  if (!is.null(args)) {
-    assert_function(derivation, names(args))
-  }
```

Comment:
I wonder if we should keep this call and update `assert_function()` such that it handles functions with `...` argument. I.e., if the functions provides the `...` argument, all arguments are accepted.

@zdz2101 , @bms63 , what do you think?

Reply:
I think the intention of the check was to ensure that all arguments specified for `args` are arguments of the function specified for `derivation`.

Reply:
Well I suppose it is a relatively short implementation, I can make an appropriate issue for it